src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php

```diff
@@ -67,7 +68,7 @@ public function process(ContainerBuilder $container): void
 
                 $this->replaceDefaultArguments([
                     0 => $id,
-                    2 => 'sonata.admin.controller.crud',
+                    2 => $defaultController,
```

Comment:
Do we have an issue here ?

On 3.x we previously inject `CRUDController::class`, now on 3.x it's `$defaultController` with the default value `CRUDController::class`.

On master the change was made to `sonata.admin.controller.crud` ; if someone write
```
services:
        app.admin.car:
            class: App\Admin\CarAdmin
            tags:
                - { name: sonata.admin, manager_type: orm, group: Demo, label: Car }
            arguments:
                - ~
                - App\Entity\Car
                - App\Controller\CarAdminController
```
it work, but what if someone wrote
```
services:
        app.admin.car:
            class: App\Admin\CarAdmin
            tags:
                - { name: sonata.admin, manager_type: orm, group: Demo, label: Car }
            arguments:
                - ~
                - App\Entity\Car
                - Sonata\AdminBundle\Controller\CRUDController
```
?

In this case the default value of the new option `default_controller` need to be changed on master.

Help @sonata-project/contributors @jordisala1991 @franmomu

Reply:
If the `default_controller` is now `sonata.admin.controller.crud`, there will be others issues.

For instance the admin maker is currently computing the controller short name this way.
```
'default_controller_short_name' => Str::getShortClassName($this->defaultController),
```

Reply:
I just tested and removing the `default_controller` feature and using `'sonata.admin.controller.crud'` back are fixing the tests.

Reply:
```
                ->scalarNode('default_controller')
                    ->defaultValue('sonata.admin.controller.crud')
                    ->cannotBeEmpty()
                    ->info('Name of the controller class to be used as a default in admin definitions')
                ->end()
```
could work.

Does it feel logic ?
And how can I get the shortname in the admin maker now ? @sonata-project/contributors

Reply:
On master we changed how we register the default controller on dependency injection. Before that change it was a Symfony controller that it was not explicitly declared on DI (the old way of registering controllers). After the change it was explicitly declared on DI: https://github.com/sonata-project/SonataAdminBundle/blob/master/src/Resources/config/core.php#L164-L167
On 3.x you won't find that definition.

IMO your change feels logic to what we did on master

Reply:
I had to change 
```
->set('sonata.admin.controller.crud', CRUDController::class)
```
to 
```
->set(CRUDController::class, CRUDController::class)
```
In the Resources/config/core.php and I don't know why (and if it's a good things).

Plus, I tried to use a custom CRUDController in the tests, and it seems like it doesn't work.
Could you help me on this ?